Enable `Math.sumPrecise` by default

Gumichocopengin8 opened the pull request:
#### b21d2072809e9d5f30797756469cec5e22fb6820
<pre>
Enable `Math.sumPrecise` by default
<a href="https://bugs.webkit.org/show_bug.cgi?id=295806">https://bugs.webkit.org/show_bug.cgi?id=295806</a>

Reviewed by Keith Miller.

All implementations and tests of `Math.sumPrecise` have been completed [1][2][3][4].
This patch enables `Math.sumPrecise` by default.
The feature flag is retained as a safeguard because the proposal is still at Stage 3 [5],
and potential issues may still arise.

[1]: <a href="https://github.com/WebKit/WebKit/commit/f49d6a41fb8d93f9c7e6758d87a06a9eea3e9698">https://github.com/WebKit/WebKit/commit/f49d6a41fb8d93f9c7e6758d87a06a9eea3e9698</a>
[2]: <a href="https://github.com/WebKit/WebKit/commit/2682d2ab17a40851f7e6bcff35126e810b0646fd">https://github.com/WebKit/WebKit/commit/2682d2ab17a40851f7e6bcff35126e810b0646fd</a>
[3]: <a href="https://github.com/WebKit/WebKit/commit/d95d6a80355e27cd4091e806e21019e4bc557a1e">https://github.com/WebKit/WebKit/commit/d95d6a80355e27cd4091e806e21019e4bc557a1e</a>
[4]: <a href="https://github.com/WebKit/WebKit/commit/480ea278f6add966e8b2c073c832989ace517bd0">https://github.com/WebKit/WebKit/commit/480ea278f6add966e8b2c073c832989ace517bd0</a>
[5]: <a href="https://github.com/tc39/proposal-math-sum/blob/4f621a637fb61baa61359e4c05e7c6d7e9629890/README.md?plain=1#L11">https://github.com/tc39/proposal-math-sum/blob/4f621a637fb61baa61359e4c05e7c6d7e9629890/README.md?plain=1#L11</a>

* LayoutTests/js/Object-getOwnPropertyNames-expected.txt:
* LayoutTests/js/script-tests/Object-getOwnPropertyNames.js:
* Source/JavaScriptCore/runtime/OptionsList.h:
* Source/WTF/wtf/PreciseSum.h:

Canonical link: <a href="https://commits.webkit.org/297839@main">https://commits.webkit.org/297839@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9373e0b3b1a9bf68fb87d073e0fddca1160046c2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/112954 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/32689 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/23167 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/119158 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/63584 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/33341 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/41252 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/85964 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/36667 "Build was cancelled. Recent messages:Printed configuration") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/115901 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/26615 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/101607 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/66270 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/25900 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/19739 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/62916 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/105471 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/96008 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/19813 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/122379 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/111570 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/40032 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/29849 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/94814 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/40416 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/97827 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/94554 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24155 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/39693 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/17506 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/36145 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/39918 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/45417 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/135800 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/39559 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/36472 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/42892 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/41296 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/4814c007adcdba412fc042266bc2764bf2aeb1c6. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48441

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/53f595f68b8568a92271bf3e18f1b2944544b1b1. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48441

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/9373e0b3b1a9bf68fb87d073e0fddca1160046c2. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48441

Gumichocopengin8 commented:
> Usually we just set the flag to `true` until the feature ships for a while then remove later. This way if something breaks it's relatively easy to turn back off.

Fixed it!

Gumichocopengin8 committed:
Enable `Math.sumPrecise` by default
https://bugs.webkit.org/show_bug.cgi?id=295806

Reviewed by Keith Miller.

All implementations and tests of `Math.sumPrecise` have been completed [1][2][3][4].
This patch enables `Math.sumPrecise` by default.
The feature flag is retained as a safeguard because the proposal is still at Stage 3 [5],
and potential issues may still arise.

[1]: https://github.com/WebKit/WebKit/commit/f49d6a41fb8d93f9c7e6758d87a06a9eea3e9698
[2]: https://github.com/WebKit/WebKit/commit/2682d2ab17a40851f7e6bcff35126e810b0646fd
[3]: https://github.com/WebKit/WebKit/commit/d95d6a80355e27cd4091e806e21019e4bc557a1e
[4]: https://github.com/WebKit/WebKit/commit/480ea278f6add966e8b2c073c832989ace517bd0
[5]: https://github.com/tc39/proposal-math-sum/blob/4f621a637fb61baa61359e4c05e7c6d7e9629890/README.md?plain=1#L11

* LayoutTests/js/Object-getOwnPropertyNames-expected.txt:
* LayoutTests/js/script-tests/Object-getOwnPropertyNames.js:
* Source/JavaScriptCore/runtime/OptionsList.h:
* Source/WTF/wtf/PreciseSum.h:

Canonical link: https://commits.webkit.org/297839@main

webkit-commit-queue commented:
Committed 297839@main (b21d2072809e): <https://commits.webkit.org/297839@main>

Reviewed commits have been landed. Closing PR #48441 and removing active labels.
